app/lib/Plugins/SearchEngine/Elastic8ConfigurationSettings.php

```diff
@@ -126,55 +107,24 @@ public function getSettingHint($pn_setting_num) {
 		return $this->setting_hints[$pn_setting_num];
 	}
 
-	/**
-	 * @throws Zend_Http_Client_Exception
-	 */
 	private function _checkElasticSearchRunning(): int {
-		/* check if elasticsearch alive */
-		$http_client = new Zend_Http_Client();
-		$http_client->setUri(
-			$this->elasticsearch_base_url . "/" . /* general url */
-			$this->elasticsearch_index_name . "/" . /* index name */
-			"/_search"
-		);
-		$http_client->setParameterGet('q', '*');
 		try {
-			$http_response = $http_client->request();
-		} catch (Zend_Http_Client_Exception $e) {
+			$this->elastic8->info();
+			return __CA_SEARCH_CONFIG_OK__;
+		} catch (\Exception $e) {
 			return __CA_SEARCH_CONFIG_ERROR__;
 		}
-
-		/* everything passed */
-
-		return __CA_SEARCH_CONFIG_OK__;
 	}
 
-	/**
-	 * @throws Zend_Http_Client_Exception
-	 */
-	private function _checkElasticSearchIndexExists(): int {
-		/* check if elasticsearch alive */
-		$http_client = new Zend_Http_Client();
-
-		$http_client->setUri(
-			$this->elasticsearch_base_url . "/" . /* general url */
-			$this->elasticsearch_index_name . "/" . /* index name */
-			"/_search"
-		);
-		$http_client->setParameterGet('q', '*');
+	private function _checkElasticSearchIndexesExist(): int {
 		try {
-			$http_response = $http_client->request();
-			$response = json_decode($http_response->getBody(), true);
-
-			if (isset($response["status"]) && $response["status"] == 404) {
-				return __CA_SEARCH_CONFIG_ERROR__;
+			if ($this->elastic8->checkIndexes()) {
+				return __CA_SEARCH_CONFIG_OK__;
 			}
-		} catch (Zend_Http_Client_Exception $e) {
+		} catch (\Exception $e) {
```

Comment:
I believe there's an ElasticsearchException interface that you can user here instead. Would be handy to display or log these errors rather than just 'hey there's something bad - you go figure it out'.

Reply:
Conceptually agree, can't quite wrap my head around what we want to do here, might need to walk through how we want to handle this with you.

Reply:
As per discussion the error I am catching here is actually a Curl error when host is not found.